modules/user_accounts/php/NDB_Menu_Filter_user_accounts.class.inc

```diff
@@ -48,13 +48,13 @@ class NDB_Menu_Filter_User_Accounts extends NDB_Menu_Filter
     {
         $user =& User::singleton();
         // the base query
-        $query = " FROM users 
+        $query = " FROM users
             LEFT JOIN user_psc_rel ON (user_psc_rel.UserID=users.ID)
             LEFT JOIN psc ON (user_psc_rel.CenterID=psc.CenterID)
             WHERE 1=1";
         if (!$user->hasPermission('user_accounts_multisite')) {
-            $site_arr     = implode(",", $user->getCenterIDs());
-            $this->query .= " AND user_psc_rel.CenterID IN (" . $site_arr . ")";
+            $site_arr = implode(",", $user->getCenterIDs());
+            $query   .= " AND FIND_IN_SET(user_psc_rel.CenterID,'$site_arr')";
```

Comment:
Can you explain how this is different and why one works and the other doesn't?

They look like they're logically the same thing to me, so I don't see why this would fix the bug.